include/machine.h

```diff
@@ -47,13 +56,31 @@ static inline region_t CONST paddr_to_pptr_reg(const p_region_t p_reg)
     };
 }
 
-static inline p_region_t CONST pptr_to_paddr_reg(const region_t reg)
+static inline p_region_t CONST pptr_to_paddr_reg(const region_t pptr_reg)
+{
+    return (p_region_t) {
+        .start = pptr_to_paddr((const void *)pptr_reg.start),
+        .end   = pptr_to_paddr((const void *)pptr_reg.end),
+    };
+}
+
+/* Returns the physical region of the kernel image boot part, ie part that is
+ * no longer needed once booting is finished. */
+static inline p_region_t get_p_reg_kernel_img_boot(void)
 {
     return (p_region_t) {
-        .start = pptr_to_paddr((const void *)reg.start),
-        .end   = pptr_to_paddr((const void *)reg.end),
+        .start = kpptr_to_paddr((const void *)KERNEL_ELF_BASE),
+        .end   = kpptr_to_paddr(ki_boot_end)
     };
 }
 
+/* Returns the physical region of the kernel image. */
+static inline p_region_t get_p_reg_kernel_img(void)
+{
+    return (p_region_t) {
+        .start = kpptr_to_paddr((const void *)KERNEL_ELF_BASE),
+        .end   = kpptr_to_paddr(ki_end)
+    };
+}
```

Comment:
Can these just return `region_t` types? Then if needed they can be wrapped in `pptr_to_paddr_reg` at any call sights? The kernel mostly deals with region_t by default.

Reply:
It look a bit odd here, but the reason for using physical addresses here is, that the whole free memory setup during boot is simpler if this all works on the physical addresses eventually. And that is also much nicer to debug then for ports or memory changes. These changes are in a draft in my fork at https://github.com/axel-h/seL4/pull/100

Reply:
Ok